src/Widgets/WindowSwitcher.vala

```diff
@@ -173,6 +182,27 @@ namespace Gala
 			dock_surface = null;
 		}
 
+		void update_background ()
```

Comment:
Afaics it does take them into account. MetaMonitorManager does which is where the size is coming from.